GDJS/Runtime/pixi-renderers/pixi-bitmapfont-manager.ts

```diff
@@ -253,50 +259,48 @@ namespace gdjs {
      * Load the "bitmapFont" resources of the game, so that they are ready
      * to be used when `obtainBitmapFont` is called.
      */
-    loadBitmapFontData(
+    async loadBitmapFontData(
       onProgress: (count: integer, total: integer) => void
-    ): Promise<void[]> {
-      const bitmapFontResources = this._resources.filter(
-        (resource) => resource.kind === 'bitmapFont' && !resource.disablePreload
-      );
-      if (bitmapFontResources.length === 0) {
-        return Promise.resolve([]);
-      }
+    ): Promise<integer> {
+      const preloadedResources = [
+        ...gdjs.filterIterable(
+          this._resources.values(),
+          (resource) => !resource.disablePreload
+        ),
+      ];
 
       let loadedCount = 0;
-      return Promise.all(
-        bitmapFontResources.map((bitmapFontResource) => {
-          return fetch(
-            this._resourcesLoader.getFullUrl(bitmapFontResource.file),
-            {
-              credentials: this._resourcesLoader.checkIfCredentialsRequired(
-                bitmapFontResource.file
-              )
-                ? // Any resource stored on the GDevelop Cloud buckets needs the "credentials" of the user,
-                  // i.e: its gdevelop.io cookie, to be passed.
-                  'include'
-                : // For other resources, use "same-origin" as done by default by fetch.
-                  'same-origin',
-            }
-          )
-            .then((response) => response.text())
-            .then((fontData) => {
-              this._loadedFontsData[bitmapFontResource.name] = fontData;
-            })
-            .catch((error) => {
-              logger.error(
-                "Can't fetch the bitmap font file " +
-                  bitmapFontResource.file +
-                  ', error: ' +
-                  error
-              );
-            })
-            .then(() => {
-              loadedCount++;
-              onProgress(loadedCount, bitmapFontResources.length);
-            });
+      await Promise.all(
+        gdjs.mapIterable(preloadedResources, async (bitmapFontResource) => {
+          try {
```

Comment:
This is an array right? Can you do a map?

```suggestion
      await Promise.all(
        preloadedResources.map(async (bitmapFontResource) => {
          try {
```